packages/itwin/property-grid/src/types.ts

```diff
@@ -8,16 +8,24 @@ import type {
   IPresentationPropertyDataProvider,
   PresentationPropertyDataProvider,
 } from "@itwin/presentation-components";
-import type { PropertyGridContextMenuArgs } from "@itwin/components-react";
+import type {
+  ActionButtonRenderer,
+  PropertyGridContextMenuArgs,
+} from "@itwin/components-react";
 import type { ContextMenuItemProps, Orientation } from "@itwin/core-react";
 import type { FavoritePropertiesScope } from "@itwin/presentation-frontend";
 import type {
+  PropertyRecord,
   StagePanelLocation,
   StagePanelSection,
 } from "@itwin/appui-abstract";
 
 export type ContextMenuItemInfo = ContextMenuItemProps &
-React.Attributes & { label: string };
+  React.Attributes & {
+    label: string;
+    hideDefaultContextMenuItem?: PropertyGridDefaultContextMenuKey;
```

Comment:
I think a better way of handling this hideDefaultContextMenuItem would be to have a way to override one or more props of any default context menu, which can be mapped by the enum you created, so that you could for instance pass an isValid handler for the favorite options

Open to suggestions, but we would want at least to ensure that each individual context menu option doesnt have its own prop to hide another, it should be a sibling of the additionContextMenuOptions

Reply:
What would need to be overridden if not visibility? Otherwise we're just creating a more complicated way to hide the menu item by overriding its isValid call to hide it when the additional context menu option is valid. I could rename it to something like replaceDefaultContextMenuItem. Otherwise, if I'm understanding correctly, it doesn't sound like that level of abstraction is needed here, since I'm not sure what other use case there is (or what other user there would be, even).

Reply:
The point is that props of a context menu item shouldn't be coupled to the unique behavior of any one context menu item. i.e. we use a `onSelect` instead of a `onShareFavorite`. Having `hideDefaultContextMenuItem` as a prop makes some specific assumptions about a particular context menu's behavior, such as that it may need to hide exactly one other context menu presumably when this context menu's isValid is true. This would need to be amended if needed to hide more than one context menu item or a context menu item that isn't a default option. It would be preferable therefore to find a way to avoid adding an additional prop that caters only to a specific use case. IMO the solution is to have a way to override default context menu item attributes as this is generally useful and provides an intuitive way to express 'this context menu cant show for the same reason this other one should show' without adding any new concept. 

It also stands to reason that a consumer may wish to modify something about a default component without having to make that change here as well, so it solves your problem and potentially a future problem

Reply:
This is pushed if you want to give it a look @jason-crow

Reply:
this is pushed if you want to take a look @jason-crow